woodpecker-go/woodpecker/client.go

```diff
@@ -116,16 +125,25 @@ func (c *client) do(rawURL, method string, in, out any) error {
 func (c *client) open(rawURL, method string, in any) (io.ReadCloser, error) {
 	uri, err := url.Parse(rawURL)
 	if err != nil {
-		return nil, err
+		return nil, &ClientError{
+			StatusCode: http.StatusInternalServerError,
+			Message:    err.Error(),
+		}
```

Comment:
Such an url error is not really related to a http status. Would keep the original errors here instead of using `http.StatusInternalServerError`.

Reply:
But doesn't internal server error still fit? I mean an API call failed without a user error.

Reply:
A http call could deliver a status code. Why would `url.Parse` deliver one or is there a reason to do so? The other errors are most of the time already custom types like https://pkg.go.dev/net/url#Error with additional details. Using `ClientError` would drop those details.